semantic-reranker/build.gradle

```diff
@@ -0,0 +1,133 @@
+import org.opensearch.gradle.test.RestIntegTestTask
+
+apply plugin: 'java'
+apply plugin: 'idea'
+apply plugin: 'opensearch.opensearchplugin'
+apply plugin: 'opensearch.yaml-rest-test'
+apply plugin: 'opensearch.pluginzip'
+
+group = 'org.opensearch'
+
+def pluginName = 'search-relevance'
+def pluginDescription = 'Make Opensearch results more relevant.'
+def projectPath = 'org.opensearch'
+def pathToPlugin = 'search.relevance'
+def pluginClassName = 'SearchRelevancePlugin'
+
+publishing {
+    publications {
+        pluginZip(MavenPublication) { publication ->
+            pom {
+                name = pluginName
+                description = pluginDescription
+                licenses {
+                    license {
+                        name = "The Apache License, Version 2.0"
+                        url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
+                    }
+                }
+                developers {
+                    developer {
+                        name = "OpenSearch"
+                        url = "https://github.com/opensearch-project/search-relevance"
+                    }
+                }
+            }
+        }
+    }
+}
+opensearchplugin {
+    name pluginName
+    description pluginDescription
+    classname "${projectPath}.${pathToPlugin}.${pluginClassName}"
+    licenseFile rootProject.file('LICENSE.txt')
+    noticeFile rootProject.file('NOTICE.txt')
+}
+
+// This requires an additional Jar not published as part of build-tools
+loggerUsageCheck.enabled = false
+
+// No need to validate pom, as we do not upload to maven/sonatype
+validateNebulaPom.enabled = false
+
+buildscript {
+    ext {
+        opensearch_version = "2.1.0-SNAPSHOT"
+    }
+
+    repositories {
+        mavenLocal()
+        maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
+        mavenCentral()
+        maven { url "https://plugins.gradle.org/m2/" }
+    }
+
+    dependencies {
+        classpath "org.opensearch.gradle:build-tools:${opensearch_version}"
+    }
+}
+
+repositories {
+    mavenLocal()
+    maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
+    mavenCentral()
+    maven { url "https://plugins.gradle.org/m2/" }
+}
+
+dependencies {
+    implementation 'com.google.guava:guava:21.0'
```

Comment:
The uses of Guava in SlidingWindowTextSplitter and SearchRelevancePlugin seem trivial enough to implement with built-in Java classes, so I would drop the Guava dependency.

The replacements needed are:
```
Lists.newArrayList()     ->    new ArrayList<>()
ImmutableList.of(...)    ->    Arrays.asList(...)
```

Reply:
Removed Guava. Will likely remove lang3 in a follow up PR too.